### PR TITLE
[#56] only compare exposed modules

### PR DESCRIPTION
**Pull request by gelisam**

### CODE REVIEW

File: src/Policeman/Diff.hs [hunk 1]
```diff
@@ -30,15 +30,21 @@ comparePackageStructures psPrev psCur = PackageDiff {..}
...
         go hm (modul, msPrev) = case HM.lookup modul (psModulesMap psCur) of
             Just msCur -> HM.insert modul (moduleDiffExport msPrev msCur) hm
             Nothing    -> hm
 
         oldModules :: [(Module, ModuleStructure)]
-        oldModules = HM.toList $ psModulesMap psPrev
+        oldModules = HM.toList (psModulesMap psPrev)
+
+        wasExposed :: Module -> Bool
+        wasExposed x = x `member` psModules psCur
```
Comment (vrom911) [inline src/Policeman/Diff.hs]: It seems, that it's better to be called `isExposed` as it checks the current exposed list, not the previous version.

File: src/Policeman/Diff.hs [hunk 2]
```diff
@@ -30,15 +30,21 @@ comparePackageStructures psPrev psCur = PackageDiff {..}
...
+
+        oldExposedModules :: [(Module, ModuleStructure)]
+        oldExposedModules = filter (wasExposed . fst) oldModules
```
Comment (vrom911) [inline src/Policeman/Diff.hs:47]: This seems to be the subset of old modules (old modules that are exported currently), is this correct?

**Review by vrom911 — COMMENTED** [review]
I am a bit confused about the mixture of filters on old modules by new modules. Could you please add more motivation in the comments for the introduced functions, and to explain it to me, please?

File: src/Policeman/Diff.hs (hunk 1 above)
Comment (gelisam) [reply src/Policeman/Diff.hs]: oops! the name is right, I meant to check the previous version's exposed list, not the current exposed list.

Comment (gelisam) [reply src/Policeman/Diff.hs]: I pushed a fix. I believe this mistake addresses all three of your comments.

File: src/Policeman/Diff.hs [hunk 3]
```diff
@@ -30,15 +30,21 @@ comparePackageStructures psPrev psCur = PackageDiff {..}
...
+    pdExport = foldl' go HM.empty oldExposedModules
       where
         go :: HashMap Module (Diff Export) -> (Module, ModuleStructure) -> HashMap Module (Diff Export)
         go hm (modul, msPrev) = case HM.lookup modul (psModulesMap psCur) of
             Just msCur -> HM.insert modul (moduleDiffExport msPrev msCur) hm
             Nothing    -> hm
 
         oldModules :: [(Module, ModuleStructure)]
-        oldModules = HM.toList $ psModulesMap psPrev
+        oldModules = HM.toList (psModulesMap psPrev)
```
Comment (gelisam) [inline src/Policeman/Diff.hs:41]: hmm, I don't remember changing this `$` to parentheses, would you like me to change it back?

Comment (vrom911) [reply src/Policeman/Diff.hs:41]: No worries, this is not a big deal 🙂

**Review by vrom911 — APPROVED** [review]
Thanks for the fix, this all makes perfect sense to me now 👍

**Pull request merged**